src/foraging_gui/Foraging.py

```diff
@@ -1216,6 +1223,105 @@ def _GetSettings(self):
         self.Other_lick_spout_distance=self.Settings['lick_spout_distance_box'+str(self.box_number)]
         self.rig_name = '{}'.format(self.current_box)
 
+    def _ConnectSlims(self):
+        '''
+            Connect to Slims
+        '''
+        try:
+            self.slims_client = SlimsClient(username=os.environ['SLIMS_USERNAME'],
+                                            password=os.environ['SLIMS_PASSWORD'])
+        except KeyError:
+            raise KeyError('SLIMS_USERNAME and SLIMS_PASSWORD do not exist as '
+                         'environment variables on machine. Please add')
+
+        try:
+            self.slims_client.fetch_model(models.SlimsMouseContent, barcode='00000000')
+        except Exception as e:
+            if str(e) != 'No record found.':    # bypass if mouse doesn't exist
+                raise Exception(f'Exception trying to read from Slims: {e}.\n'
+                                f' Please check credentials:\n'
+                                f'Username: {os.environ["SLIMS_USERNAME"]}\n'
+                                f'Password: {os.environ["SLIMS_PASSWORD"]}')
+
+
+
+    def _AddWaterLogResult(self, session: Session):
+        '''
+            Add WaterLogResult to slims based on current state of gui
+
+            :param session: Session object to pull water information from
+
+        '''
+
+        try:    # try and find mouse
+            mouse = self.slims_client.fetch_model(models.SlimsMouseContent, barcode=session.subject_id)
+        except Exception as e:
+            if str(e) == 'No record found.':    # if no mouse found or validation errors on mouse
+                # check schedule to make sure enough information is known about mouse to create model in slims
+                if not pd.isnull((curr_st := self._GetInfoFromSchedule(session.subject_id, 'Current State'))) \
+                        and not pd.isnull((point_of_contact := self._GetInfoFromSchedule(session.subject_id, 'PI'))):
+                    model = models.SlimsMouseContent(barcode=session.subject_id,
```

Comment:
Add a logging error here
logging.error('Mouse not in SLIMS, creating new entry....')